File: blocks-world/src/main/scala-2.11/com/btesila/aose/lab4/Predicate.scala
package com.btesila.aose.lab4

sealed trait Predicate

object Predicate {
  case class On(block: Block, onBlock: Block) extends Predicate
  case class OnTable(block: Block) extends Predicate
  case class Clear(block: Block) extends Predicate
  case class Hold(block: Block) extends Predicate
  case object EmptyArm extends Predicate

  case class Preconditions(predicates: List[Predicate])
}

File: blocks-world/src/main/java/com/btesila/aose/lab7r/Predicate.java
package com.btesila.aose.lab7r;

/**
 * TODO Add class description.
 *
 * @author ramona.popa@airbus.com (original version)
 * @author ramona.popa@airbus.com (responsible)
 * @since 2016.11.07 @ 11:24
 */
public abstract class Predicate {

    public abstract boolean on(Block A, Block B) ;


    public abstract boolean onTable(Block A);

    public abstract boolean  clear(Block A);


    public abstract Boolean hold(Block A);

    public abstract Boolean armEmpty();

}